### PR TITLE
feat: show contract call arguments in a more approchable way

### LINKED ISSUE #142: Show decoded values for known contract events
### Please describe the problem that your request should resolve.

Contract details arguments are encoded.

### Describe the solution you'd like
Show decoded values for known contract events, instead of encoded values. 

### Is it already possible to achieve the same outcome in any other way? If so, how?
No

### Additional context

image

### PR DESCRIPTION
**Pull request by lukeromanowicz**
## Description
Resolves #142 

MAINNET:
Example of AEX-9 contract with non-decoded arguments: ct_2Hyt9ZxzXra5NAzhePkRsDPDWppoatVD7CtHnUoHVbuehwR8Nb
Example of contract with decoded arguments: ct_2U1usf3A8ZNUcZLkZe5rEoBTxk7eJvk9fcbRDNqmRiwXCHAYN

TESTNET:
Example of minting NFTs: ct_cVbugNVmdz2e99bu7fiBHsJY1NhRHh53CQBjNTdaxh8pSKHgK
Example of minting NFTs using templates: ct_Wr8AH1LDaya9niRfH92atd8LerNUYVcFdYrvcTdriMrqxNzvs

## Demo
https://github.com/aeternity/aescan/assets/46789227/1ce5b57a-546c-484c-b906-ff36ddee5e6f

AEX-141 tokens (these tokens by design don't have symbols - only id, that's why it's not present in the chips):
image
image

## Checklist:

- [x] I have read and followed the [Contributing Guide](../../CONTRIBUTING.md)  
- [x] My change does not require a change to the documentation.

### CODE REVIEW

File: src/app.vue
Comment (lukeromanowicz) [inline src/app.vue:101]: unrelated improvement to show error details in the console

**Review by michele-franchi — COMMENTED** [review]
I had a quick look at the demo and I noticed known AEX-N events are not displayed. If you check this comment https://github.com/aeternity/aescan/issues/142#issuecomment-1693056178 one of the requirements is to also display known AEX-N events in the usual way with the mix of expandable buttons for events that cannot be decoded.

File: src/components/ContractEventDataPanel.vue
Comment (janmichek) [inline src/components/ContractEventDataPanel.vue:8]:
```suggestion
        v-for="(arguments, index) in event.data"
```

I would personally better read arguments than arg, if there are no technical limitations for shortening

**Review by janmichek — CHANGES_REQUESTED** [review]
Code looks good and works nice as well 👍 

When I look at the design, there are few misalignemnts

image

1. the copy button is not implemented. Is this intentional? Please add it or negotiate the change in figma
2. the styles of expanded rows differs (row spacingand border radius). I better like our version. Can you negotiate the change?
3. The button label differs. Can you negotiate the change?

**Comment by lukeromanowicz** [comment]
@michele-franchi thank you for your review. I implemented the missing features of showing decoded values for known events.

@janmichek Very good eye! I added the missing copy feature and changed the existing code to reflect the recent design changes. Unfortunately some of the changes are inconsistent with the rest of the application so I suggested some improvements in the designs. So far no response but I have reasonable confidence that they will be eventually approved. We can wait until they are approved or open another task if they won't.

File: src/components/ContractEventsTable.vue
Comment (michele-franchi) [inline src/components/ContractEventsTable.vue:48]:
```suggestion
          {{ event.eventName || 'N/A' }}
```

**Review by michele-franchi — APPROVED** [review]
I found some minor issues but it looks good, I approve in advance.

File: src/components/ContractEventDataPanel.vue
Comment (lukeromanowicz) [reply src/components/ContractEventDataPanel.vue]: Right, fixed :+1:

**Comment by CharisSiarampalis** [comment]
@lukeromanowicz When i try to click on some smart contracts i can see the page loading up but then nothing appears
Example:
https://pr-486-aescan.stg.aepps.com/contracts/ct_rtqLKcZfQSx1GjUVBkVtRjmTGf3E2DLrSYYcaQsNYNrEtTyN1
image

On dev  i can see it
https://aescan.dev.aepps.com/contracts/ct_rtqLKcZfQSx1GjUVBkVtRjmTGf3E2DLrSYYcaQsNYNrEtTyN1

**Comment by lukeromanowicz** [comment]
@CharisSiarampalis that's an interesting case. It's not an AEX-x compliant token and yet it's decoded. Weird. Nonetheless fixed. Thanks for reporting

**Comment by CharisSiarampalis** [comment]
@lukeromanowicz In some cases i see N/A for name.
Is this normal?
image

**Comment by lukeromanowicz** [comment]
@CharisSiarampalis correct, not all of the events can be decoded by the middleware

**Review by CharisSiarampalis — APPROVED** [review]

File: src/components/ExpandButton.vue
Comment (janmichek) [inline src/components/ExpandButton.vue:2]: is the `div` wrap needed?

**Review by janmichek — CHANGES_REQUESTED** [review]
Nice work! User-wise I am satisfied. Giving some code suggestions. Let me know what you think.

**Comment by janmichek** [comment]
The copy icons does not seem to be aligned to anything
image

**Comment by janmichek** [comment]
image

Probably not related to the feature, but could you add some whitespace  between the last row and pagination? Seems to be inconsistent with other tables

**Comment by janmichek** [comment]
Could you utilize better the space for arguments value here?

image

File: src/components/ContractEventCell.vue
Comment (lukeromanowicz) [reply src/components/ContractEventCell.vue:35]: Since computed property should not modify another variable when getting read, introducing hasError here would be against good practices of vue clean code. We would either need to catch this error in the parent components (where ContractEventCell is used) or define a new vue component that in the template simply shows <span>N/A</span>. I considered all of these options and thought that the currently used approach makes the most sense and is simplistic.

File: src/components/ContractEventCellTemplateCreation.vue
Comment (lukeromanowicz) [reply src/components/ContractEventCellTemplateCreation.vue:7]: Some of the Event Cells need it. If I remove this in the cells when this is unused, vue will start sending warnings that a value was passed to a property that doesn't exist :/

File: src/components/ExpandButton.vue
Comment (lukeromanowicz) [reply src/components/ExpandButton.vue:2]: Absolutely not. Thanks :+1:

**Comment by lukeromanowicz** [comment]
Very good eye and great suggestions @janmichek! Thank you! I've applied all of the recommendations except those on which I commented. Please let me know what you think

**Review by janmichek — CHANGES_REQUESTED** [review]
I found one more issue. See this scenario

https://github.com/aeternity/aescan/assets/15363559/4ac46959-d2f0-451e-8578-59d4a92eeb9a

File: src/components/ContractEventCellTemplateCreation.vue
Comment (janmichek) [reply src/components/ContractEventCellTemplateCreation.vue:7]: I see, but why is this labeled as `required` then?

File: src/components/ContractEventCell.vue
Comment (janmichek) [reply src/components/ContractEventCell.vue:35]: Thanks for an explanation, I think it makes sense. Could you point me to that good practices of vue clean code?

**Comment by lukeromanowicz** [comment]
@janmichek it's a middleware issue throwing 500. We are (not) handling it exactly the same in every other table of the app. How do you suggest to fix it?

**Comment by janmichek** [comment]
> @janmichek it's a middleware endpoint issue which is throwing 500. We are (not) handling it exactly the same in every other table of the app. How do you suggest to fix it?

If it's a middleware issue, then I would report it and create some followup on our side

**Comment by lukeromanowicz** [comment]
Reported to the MDW and created a followup on our side: https://github.com/aeternity/aescan/issues/498

**Review by janmichek — APPROVED** [review]

**Pull request merged**
